cloud/jenkins/ps_operator_eks.groovy

```diff
@@ -0,0 +1,299 @@
+                        else
+                            cd ./source/
+                            sg docker -c "
+                                docker login -u '${USER}' -p '${PASS}'
+                                export IMAGE=perconalab/percona-server-mysql-operator:$GIT_BRANCH
+                                ./e2e-tests/build
+                                docker logout
+                            "
+                            sudo rm -rf ./build
+                        fi
+                    '''
+                }
+            }
+        }
+        stage('Create EKS Infrastructure') {
+            steps {
+                sh '''
+cat <<-EOF > cluster.yaml
+# An example of ClusterConfig showing nodegroups with mixed instances (spot and on demand):
+---
+apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+
+metadata:
+    name: eks-psmo-cluster
+    region: eu-west-3
+    version: "$PLATFORM_VER"
+
+nodeGroups:
+    - name: ng-1
+      minSize: 3
+      maxSize: 5
+      instancesDistribution:
+        maxPrice: 0.15
+        instanceTypes: ["m5.xlarge", "m5.2xlarge"] # At least two instance types should be specified
+        onDemandBaseCapacity: 0
+        onDemandPercentageAboveBaseCapacity: 50
+        spotInstancePools: 2
+      tags:
+        'iit-billing-tag': 'jenkins-eks'
+EOF
+                '''
+
+                withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: 'eks-cicd', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+                     sh """
+                         export PATH=/home/ec2-user/.local/bin:$PATH
+                         source $HOME/google-cloud-sdk/path.bash.inc
+                         eksctl create cluster -f cluster.yaml
+                     """
+               }
+               stash includes: 'cluster.yaml', name: 'cluster_conf'
+
+            }
+        }
+        stage('E2E Basic Tests') {
+            options {
+                timeout(time: 3, unit: 'HOURS')
+            }
+            steps {
+                runTest('init-deploy')
+                runTest('monitoring')
+                runTest('semi-sync')
+                runTest('sidecars')
+            }
+        }
+        stage('Make report') {
+            steps {
+                makeReport()
+                sh """
+                    echo "${TestsReport}" > TestsReport.xml
+                """
+                step([$class: 'JUnitResultArchiver', testResults: '*.xml', healthScaleFactor: 1.0])
+                archiveArtifacts '*.xml'
+            }
+        }
+    }
+
+    post {
+        always {
+                withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: 'eks-cicd', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+                    unstash 'cluster_conf'
+                    sh """
+                        eksctl delete cluster -f cluster.yaml --wait
```

Comment:
I propose you add `--force` here, because if it will fail to drain the pods from the nodes it will not delete AWS resources.